### PR TITLE
Drop Cockpit dependency for web translations

### PR DESCRIPTION
**Pull request by lslezak**
## Problem

- Drop Cockpit dependency for loading the web translations

## Solution

- Reimplement translation loading without Cockpit

## Notes

- The code still works with the Cockpit backend and can be merged to `master` without breaking anything.
- It uses the same process and same translation loading mechanism as Cockpit just without using the Cockpit library. This makes the transition easy and simple.
- In the future we can optionally change that (and even use a different framework like i18n-next).
- There are some comments marking the code for removal when the Cockpit dependency is fully dropped.

## Testing

- Updated unit tests
- Tested manually

## Screenshots

It still works! :smiley: 
agama_spanish

### CODE REVIEW

File: web/webpack.config.js [hunk 1]
```diff
@@ -46,7 +46,9 @@ const copy_files = [
 const plugins = [
   new Copy({ patterns: copy_files }),
   new Extract({ filename: "[name].css" }),
-  new CockpitPoPlugin(),
+  // the wrapper sets the main code called in the po.js files,
+  // the PO_DATA tag is replaced by the real translation data
+  new CockpitPoPlugin({ wrapper: "agama.locale(PO_DATA);" }),
```
Comment (lslezak) [inline web/webpack.config.js:51]: Note: The default wrapper is `agama.locale(PO_DATA)` which calls the Cockpit implementation.

File: web/src/context/installerL10n.jsx [hunk 2]
```diff
@@ -49,37 +50,43 @@ function useInstallerL10n() {
...
+ * It takes the language from the AgamaLang cookie.
  *
  * @return {string|undefined} Undefined if language is not set.
  */
-function cockpitLanguage() {
+function agamaLanguage() {
   // language from cookie, empty string if not set (regexp taken from Cockpit)
   // https://github.com/cockpit-project/cockpit/blob/98a2e093c42ea8cd2431cf15c7ca0e44bb4ce3f1/pkg/shell/shell-modals.jsx#L91
-  const languageString = decodeURIComponent(document.cookie.replace(/(?:(?:^|.*;\s*)CockpitLang\s*=\s*([^;]*).*$)|^.*$/, "$1"));
+  const languageString = decodeURIComponent(document.cookie.replace(/(?:(?:^|.*;\s*)AgamaLang\s*=\s*([^;]*).*$)|^.*$/, "$1"));
```
Comment (imobachgs) [inline web/src/context/installerL10n.jsx:62]: I am not sure about the best practices for naming cookies. We used `agamaToken` for the token, following the example we found in the [Set-Cookie documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#session_cookie) (e.g., `sessionId`).

I would try to be consistent.

File: web/src/context/installerL10n.jsx [hunk 3]
```diff
@@ -49,37 +50,43 @@ function useInstallerL10n() {
...
+function storeAgamaLanguage(language) {
+  const current = agamaLanguage();
   if (current === language) return false;
 
   // Code taken from Cockpit.
-  const cookie = "CockpitLang=" + encodeURIComponent(language) + "; path=/; expires=Sun, 16 Jul 3567 06:23:41 GMT";
+  const cookie = "AgamaLang=" + encodeURIComponent(language) + "; path=/; expires=Sun, 16 Jul 3567 06:23:41 GMT";
   document.cookie = cookie;
+
+  // for backward compatibility, CockpitLang cookie is needed to load correct po.js content from Cockpit
```
Comment (imobachgs) [inline web/src/context/installerL10n.jsx:84]: :+1:

**Review by imobachgs — COMMENTED** [review]
It looks good. Just a comment about the cookie name.

**Review by imobachgs — APPROVED** [review]

**Pull request merged**